frontend/catalyst/jax_primitives.py

```diff
@@ -294,7 +294,9 @@ def _python_callback_lowering(
     ip = module.body
     attrs = [fn_ty_attr, callback_id, len(args), len(results_ty)]
     with ir.InsertionPoint(ip):
-        callbackOp = CallbackOp(f"callback_{callback_id}", *attrs)
+        # TODO: Name mangling for callbacks
+        name = callback.__name__
```

Comment:
Why not using `callback.__name__` directly?

Reply:
Maybe it is a good idea! I wanted to preserve some indication that it is a callback at the callsite. This is to distinguish a callback from a regular function that may exist in the IR. If you prefer I can make the change :)

Reply:
https://github.com/PennyLaneAI/catalyst/pull/919/files#r1671225872